src/ngraph/runtime/cpu/mkldnn_emitter.cpp

```diff
@@ -138,6 +138,1113 @@ std::vector<size_t>& MKLDNNEmitter::get_primitive_deps(size_t index)
+        md.layout_desc.blocking.padding_dims[i] = dim[i];
+        md.layout_desc.blocking.offset_padding_to_data[i] = 0;
+        md.dims[i] = dim[i];
+    }
+    md.layout_desc.blocking.offset_padding = 0;
+
+    return mkldnn::memory::desc(md);
+#endif
+}
+
+size_t MKLDNNEmitter::build_quantize_reorder(const mkldnn::memory::desc& input_desc,
+                                             const mkldnn::memory::desc& result_desc,
+                                             const std::vector<float>& scales)
+{
+    mkldnn::primitive_attr attr;
+    attr.set_output_scales(0, scales);
+    size_t input_index, result_index, primitive_index;
+
+#if defined(USE_MKLDNN_V1)
+    input_index = build_memory(input_desc);
+    result_index = build_memory(result_desc);
+    auto reorder_prim_desc = mkldnn::reorder::primitive_desc(
+        executor::global_cpu_engine, input_desc, executor::global_cpu_engine, result_desc, attr);
+    primitive_index = insert_primitive(new mkldnn::reorder(reorder_prim_desc));
+#else
+    input_index = build_memory_primitive(input_desc);
+    result_index = build_memory_primitive(result_desc);
+
+    attr.set_int_output_round_mode(mkldnn::round_mode::round_nearest);
+    auto reorder_desc = mkldnn::reorder::primitive_desc({input_desc, executor::global_cpu_engine},
+                                                        {result_desc, executor::global_cpu_engine},
+                                                        attr);
+    primitive_index = insert_primitive(new mkldnn::reorder(
+        reorder_desc, *m_mkldnn_primitives[input_index], *m_mkldnn_primitives[result_index]));
+#endif
+
+    NGRAPH_CHECK(m_primitive_deps.find(primitive_index) == m_primitive_deps.end(),
+                 "Dependencies already created for node");
+
+    m_primitive_deps[primitive_index] = {input_index, result_index};
+    return primitive_index;
+}
+
+size_t MKLDNNEmitter::build_dequantization(const ngraph::Node* node,
+                                           const mkldnn::memory::desc& input_desc,
+                                           const mkldnn::memory::desc& result_desc)
+{
+    auto dequantize = static_cast<const ngraph::op::Dequantize*>(node);
+    auto scale_const_op =
+        std::dynamic_pointer_cast<ngraph::op::Constant>(dequantize->get_argument(1));
+    std::vector<float> scale = {1.0f};
+    if (scale_const_op != nullptr)
+    {
+        scale = scale_const_op->get_vector<float>();
+    }
+    std::vector<float> scales;
+    scales.push_back(scale[0]);
+    size_t dequantize_index = 0;
+    dequantize_index = this->build_quantize_reorder(input_desc, result_desc, scales);
+    return dequantize_index;
+}
+
+size_t MKLDNNEmitter::build_reorder(const mkldnn::memory::desc& input_desc,
+                                    const mkldnn::memory::desc& result_desc)
+{
+    size_t input_index, result_index, primitive_index;
+
+#if defined(USE_MKLDNN_V1)
+    input_index = build_memory(input_desc);
```

Comment:
identical code in #if and #else ?

Reply:
one is build_memory and another one is build_memory_primitive. In mkl_dnn v1.0, mkldnn::memory is no longer primitive.